packages/genesys-spark-components/src/components/beta/gux-avatar/gux-avatar.tsx

```diff
@@ -61,27 +100,99 @@ export class GuxAvatar {
     }
   }
 
-  componentWillLoad(): void {
+  private renderBadge(): JSX.Element | null {
+    if (this.notifications) {
+      return this.renderNotificationsBadge();
+    } else if (this.badge && this.presence !== 'idle') {
+      return (
+        <div
+          class={{
+            'gux-avatar-badge': true,
+            [`gux-${this.presence}`]: true,
+            [`gux-${this.size}`]: true
+          }}
+        >
+          <gux-icon icon-name={this.getPresenceIcon(this.presence)}></gux-icon>
+        </div>
+      ) as JSX.Element;
+    }
+  }
+
+  private renderNotificationsBadge(): JSX.Element | null {
+    return (
+      <div
+        class={{
+          'gux-avatar-badge gux-notifications': true,
+          [`gux-${this.size}`]: true
+        }}
+      >
+        <gux-icon icon-name="fa/bell-regular"></gux-icon>
+      </div>
+    ) as JSX.Element;
+  }
+
+  private getPresenceIcon(presence: GuxAvatarPresence): string {
+    switch (presence) {
+      case 'available':
+        return 'fa/circle-check-solid';
+      case 'busy':
+      case 'meeting':
+        return 'fa/ban-outline';
+      case 'away':
+      case 'break':
+      case 'meal':
+      case 'training':
+        return 'fa/clock-outline';
+      case 'on-queue':
+        return 'fa/headset-solid';
+      case 'offline':
+        return 'fa/circle-xmark-regular';
+      case 'out-of-office':
+        return 'fa/subtract-circle';
+      default:
+        return 'fa/circle-xmark-regular';
+    }
+  }
+
+  private getDescriptionText(): string {
+    if (this.notifications)
+      return `${this.name} (${this.i18n('notifications')})`;
+    if (this.label) return `${this.name} (${this.label})`;
```

Comment:
I also think this should be gated by `if (this.presence !== 'none')` as well as `this.label`